test/bootlogictest_gd.cpp

```diff
@@ -0,0 +1,227 @@
+#include "CppUTest/TestHarness.h"
```

Comment:
Again, I don't see the point of having two test files, because the logic under test is exactly the same. In contrast, a test of the actual copy function is missing. Another useful test would be to check if you didn't accidentally overwrite something outside the target app's range.

Reply:
Okay that makes sense, I've condensed them into a single file. How would you test the copy function without an actual large block of flash to copy? Would you just check that the last programmed address does not extend beyond the expected range?

Reply:
So after reviewing again, I noticed you implemented the actual copy function in a platform dependent way (which makes perfect sense), so there is probably no point in writing native unit tests. It's a rather simple function, I'm fine with not having a unit test for it. Up to you if you want to write one or just leave it as it is.